syne_tune/optimizer/schedulers/synchronous/hyperband_impl.py

```diff
@@ -61,9 +63,9 @@ class SynchronousGeometricHyperbandScheduler(SynchronousHyperbandScheduler):
         :class:`FIFOScheduler`. Must be positive int larger than
         `grace_period`. If this is not given, it is inferred like in
         :class:`FIFOScheduler`.
-    brackets : int
-        Number of brackets to be used. The value 1 corresponds to successive
-        halving. Is capped to the largest number of supported brackets.
+    brackets : int (optional)
+        Number of brackets to be used. The default is to use the maximum
```

Comment:
what is "maximum number of brackets"? this seems to be a breaking API change. Is there a way how we communicate this?

Reply:
I think this just changed the docstring to make it exact. Also in the paper, if you do not pass brackets, it will be set to the max value possible, which is the total number of rungs.

The first bracket always contains all rungs, and subsequent ones always have one rung less (they drop the front ones).

This is the default for *synchronous* HB. For async HB, the default is to have 1 bracket. This is because empirically, async works quite well with 1 bracket (even though maybe Hyper-Tune may do better), but sync HB works better with >1 bracket.

Reply:
In L35 you removed the default bracket value of 1, now changing it to a different value. I'm all open for that but from my own experience it is a bit annoying if defaults keep changing.

Reply:
You are right, I was missing that.